R/basic.R

```diff
@@ -22,6 +25,15 @@ amqp_publish <- function(conn, body, exchange = "", routing_key = "",
   if (!inherits(conn, "amqp_connection")) {
     stop("`conn` is not an amqp_connection object")
   }
+  if (inherits(body, "character")) {
+    if (length(body) > 1) {
+      stop("`body` argument should be 'raw' vector or character of length 1 (string)")
+    }
+    body = charToRaw(body)
+  }
+  if (!inherits(body, "raw")) {
```

Comment:
I may be wrong, but I'm not sure this check is needed -- doesn't `RAW()` throw an error at the C level if we pass a different type?